### PR TITLE
[FIX] web: show debug tooltip on nolabel fields

### PR DESCRIPTION
**Pull request by ushyme**
## Steps to reproduce
* Activate debug mode
* Go to a form view that has a field without label (e.g., INTERNAL NOTES on a product's form view)
* hover the mouse over the field

The debug tooltip does not appear.

opw-3047994

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
seems to work ok

I'm not sure where `showTooltip` is filled or not (only times I see it set is from form_controller.js but field with tooltip don't match that). As a clean fix, I think it might be better to have `this.props.showTooltip` set to true instead of fixing it here.

**Comment by Yenthe666** [comment]
This is going to make our team very happy 😉 cc @sswapnesh

**Comment by ushyme** [comment]
@odoo/rd-framework-js

**Comment by aab-odoo** [comment]
@jpp-odoo could you take a look at this next week? Writing on the props is probably not the best way to do...

**Review by jpp-odoo — CHANGES_REQUESTED** [review]
This commit go against the main idea of the commit : f981c5f8f09ff884f0e134ae8d4ee0ad5f56e3f4

The main idea of the commit was to remove all the technical hovers and put it on an icon next to the labels. This was done to reduce the hover zones and to be more discoverable (to have an icon to show).

Also, the commit introduces two different way to have the technical information, by hover the field and by hover the '?' icon, and generally speaking having two different behaviors on the same page is not a good idea.

I agree with @simongoffin , IMHO this is more a limitation of the new tooltip than a bugfix. But, I think that maybe @bouvyd may have a good idea on how to fix this.

File: addons/web/static/src/views/fields/field.js [hunk 1]
```diff
@@ -80,6 +80,9 @@ export class Field extends Component {
             const fieldType = this.props.record.fields[this.props.name].type;
             this.FieldComponent = getFieldClassFromRegistry(fieldType, this.props.type);
         }
+        if (typeof this.props.showTooltip == "undefined") {
+            this.props.showTooltip = this.props.fieldInfo.noLabel;
+        }
```
Comment (jpp-odoo) [inline addons/web/static/src/views/fields/field.js:85]: Generally speaking, it is always better to avoid writing on the props. The idea is to leave the props as the original information received from the parent.

In this case, a better fix will be : 

```suggestion
       this.showTooltip = this.props.showTooltip;
        if (typeof this.showTooltip == "undefined") {
            this.showTooltip = this.props.fieldInfo.noLabel;
        }
```

And modify in `get tooltip(...` to use `this.showTooltip` and not `this.props.showTooltip`.

But again, I'm not sure that this is the best fix, or if it needed to be fixed.

**Comment by bouvyd** [comment]
Meh, the initial commit is the kind of quick decision that makes sense for 90% of cases out-of-the-box, but not for all cases - I don't mind cleaning up later on and deciding to cover a bit more than what was intended.


However, I agree with @jpp-odoo that the initial goal was to avoid popping things left and right as you move your cursor around and that this commit does not take that into account
=> make sure that this behaviour is only applied **_when in debug_**.

That means that in debug you have your tooltips for devs, and otherwise you don't have tooltips activated by huge areas for users not in dev mode.

**Comment by jpp-odoo** [comment]
@ushyme @nle-odoo I propose this fix : https://github.com/odoo/odoo/compare/16.0...odoo-dev:odoo:16.0-tooltip_nolable-opw-3047994-jpp

The difference is that we add in the compilation time the `showTooltip` if there is no label. Doing like this we receive the correct props from the beginig.

@bouvyd , I test it, it will only show the technical information in `debug` mode.

**Comment by bouvyd** [comment]
Hello,

After discussion with @jpp-odoo, we both agree not to fix this. Indeed, fixing this would require code spaghettification (because right now tooltip for fields are the responsibility of the labels, and trying to get tooltips without them would require code duplication and hacky-hacky behaviours code-wise) and is not worth it in our opinions, especially for a "comfort" feature meant for devs.

In short: before 16.0, tooltips were tied to fields. Starting from 16.0 onwards, tooltips are tied to labels => **no label, no tooltip**.

For devs, there are other ways to obtain this information:
- Debug > Get View
- Debug > View Fields
- Browser console => network => "get_views" call

Have a good day y'all!

**Comment by Yenthe666** [comment]
🤦 I can't believe this is the final take. This worked great in previous versions and is now semi-broken in V16.
Who was even remotely thinking it would be a good idea to bind it to labels? There's dozens of cases with fields not having labels. Was this even discussed/reviewed/tested with other parties who do day to day development and/or projects? This is a serious step back and I'd like to ask you to re-evaluate this take. Especially for (semi)-functional consultants this is an extra pain.

**Comment by bouvyd** [comment]
I'm not surprised you disagree, because there is always disagreement on small details like this one.

Yes, we tested and discussed it - and we put more emphasis on the experience of the user in the end.

What this change solved:
- it was not possible to know whether fields had a tooltip or not (you had to hover on the field or the label to know)
- tooltips where not available on mobile
- having tooltips appear on hover could be badly adapted to some field type (e.g. html field, where the content itself could have `title` attributes)
- moving your mouse anywhere in dev mode would trigger a boatload of popovers half of the time than were in the way quite often

We decided to improve the user experience and move the tooltips on the field labels, because it simply makes sense in web UI in 2022. Do you expect a note field to display a help message explaining how it works in online tools? Form tooltips are usually on labels, not on the input.

Having multiple mechanisms to display those would lead to more complex code, more maintenance, more bugs, etc. We opted for the simple solution: no label, no tooltip. Bikeshedding on this would be a waste of resources.

We have to juggle with priorities, decide what is worth it or not... You may not like it and facepalm us, you may think it's dumb - the simple fact is that we have to make choices and when the choice is between the user experience and the dev/consultant's comfort, i'll choose the former nearly every time - esp. when the loss is small.

You might also argue that we could have made other choices that would have led to UX improvement without dev mode feature loss, but the feature would have been more complex (both in ui and in code) and longer to develop. We decided it was not worth it, and I don't regret this decision at all.
